api/lib/opentelemetry/metrics/instruments.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Metrics
+    # The user-facing metrics API supports producing diagnostic measurements
+    # using three basic kinds of instrument. "Metrics" are the thing being
+    # produced -- mathematical, statistical summaries of certain observable
+    # behavior in the program. "Instruments" are the devices used by the
+    # program to record observations about their behavior. Therefore, we use
+    # "metric instrument" to refer to a program object, allocated through the
+    # API, used for recording metrics. There are three distinct instruments in
+    # the Metrics API, commonly known as Counters, Gauges, and Measures.
+    module Instruments
+      # A float gauge instrument.
+      class FloatGauge
+        def set(value, label_set_or_label = nil, *labels); end
```

Comment:
Are the `label_or_label_set` and `*labels` arguments here for overloading `set` to work with the [label set calling convention](https://github.com/open-telemetry/opentelemetry-specification/blob/5802f6e33644c5996c2fb46ef9f444f81ac2b158/specification/api-metrics-user.md#label-set-calling-convention) and the [option to bypass meter.Lables()`?

Reply:
Yep. I'm open to splitting the methods for each calling convention.

Reply:
I suppose we can see how the implementation looks. We can leave them as is for now and decide to change them if it makes sense.

Reply:
It turned out this was wrong. I've fixed in recent commits.